Add docstrings to `Wallet`

reez opened the pull request:
<!-- Erase any parts of this template not applicable to your Pull Request. -->

### Description

The [UDL file supports docstring comments](https://mozilla.github.io/uniffi-rs/udl/docstrings.html#docstrings).

Adds docstrings to `Wallet`

Supersedes #454 

### Notes to the reviewers

We can do a mega-PR, but this PR just takes a section of the UDL (separate PR's for each section).

Per 1.0.0 beta 5 https://docs.rs/bdk_wallet/latest/bdk_wallet/index.html

_Notable, can be updated in this PR:_
- Didn't include `examples` in code comments
- Didn't add code comments for constructors or `persist` or top-level `Wallet`

_Follow up_
- [ ] Add checklist item about adding docstrings (to PR template)
- [ ] Add section to README about docstrings (per Thunder)
- [ ] We probably want to standardize some minor formatting (that doesn't seem to effect output on Swift side though)
  - space/tabs/none
  - line breaks
  - alphabetize

### Changelog notice

<!-- Notice the release manager should include in the release tag message changelog -->
<!-- See https://keepachangelog.com/en/1.0.0/ for examples -->

### Checklists

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

#### New Features:

* [ ] I've added tests for the new feature
* [x] I've added docs for the new feature

#### Bugfixes:

* [ ] This pull request breaks the existing API
* [ ] I've added tests to reproduce the issue which are now passing
* [ ] I'm linking the issue being fixed by this PR

thunderbiscuit reviewed (changes requested):
This looks good! Might as well start merging that stuff. My only requested change would be to indent the comments to the same indentation level as the type/method they document, the way it's typically done in Rust/Python/Kotlin/Swift.

reez commented:
> This looks good! Might as well start merging that stuff. My only requested change would be to indent the comments to the same indentation level as the type/method they document, the way it's typically done in Rust/Python/Kotlin/Swift.

Done. rebased too. ready for merge after ACK

reez commented:
> > This looks good! Might as well start merging that stuff. My only requested change would be to indent the comments to the same indentation level as the type/method they document, the way it's typically done in Rust/Python/Kotlin/Swift.
> 
> Done. rebased too. ready for merge after ACK

rebased again, ready for merge after ACK

thunderbiscuit reviewed (approved):
ACK 0329ddc282f3977e245ada3f0c3f46f7bb2ddf07.
